components/butterfly/src/rumor/service.rs

```diff
@@ -166,6 +174,13 @@ impl Rumor for Service {
     fn id(&self) -> &str { &self.member_id }
 
     fn key(&self) -> &str { self.service_group.as_ref() }
+
+    fn expiration(&self) -> &RumorExpiration { &self.expiration }
+
+    fn expire(&mut self) {
+        self.expiration.expire();
+        self.incarnation += 1;
```

Comment:
If we don't increment the incarnation, then when we pass a rumor containing an expiration date to another peer, and that peer receives it, the received rumor with the expiration date won't override the existing rumor the peer has, and will instead get silently discarded by the `merge` mechanism.

Maybe this isn't the right mechanism for doing this (although if we remove rumor expiration from rumor comparisons, I suspect it probably is), but we need some way of saying "this rumor I just received that has an expiration date on it should supercede the rumor I currently have".  This was my way of doing that.

Reply:
That makes a lot of sense. I wonder if we can use the expiration itself for this. Since only the progenitor of the rumor will ever set it, it we can assume an update will alway have an expiration which is later than the original.

I'm still not sure if that logic should be a part of the merge operation on the rumors themselves, or occur at the level of the `RumorStore`, but either way, I think we'd need a similar mechanism on any rumor that can be updated or expire.

Reply:
I know the SWIM paper used `incarnation` as a mechanism for providing precedence on health updates, but the fact that we're using it on rumors unrelated to SWIM made me think that, in those cases, it was a generic mechanism for signaling that something newer happened.  This felt like it fit that criteria for me, but it's possible I'm not thinking about this correctly.